### ISSUE TITLE
Nullable properties are doubly nullable

### ISSUE THREAD

**Issue by geirsagberg** [bkbnio/kompendium#367]
**Describe the bug**
Types with nullable properties are generated with two times `oneOf`, once for the property and once for the type:

```json
      "ProfileUpdateRequest": {
        "type": "object",
        "properties": {
          "metadata": {
            "oneOf": [
              {
                "type": "null"
              },
              {
                "$ref": "#/components/schemas/ProfileMetadataUpdateRequest"
              }
            ]
          },
          ...
        },
        "required": []
      },
      "ProfileMetadataUpdateRequest": {
        "oneOf": [
          {
            "type": "null"
          },
          {
            "type": "object",
            "properties": {
              ...
            },
            "required": []
          }
        ]
      }
```

Only the property should be nullable:

```json
      "ProfileUpdateRequest": {
        "type": "object",
        "properties": {
          "metadata": {
            "oneOf": [
              {
                "type": "null"
              },
              {
                "$ref": "#/components/schemas/ProfileMetadataUpdateRequest"
              }
            ]
          },
          ...
        },
        "required": []
      },
      "ProfileMetadataUpdateRequest": {
        "type": "object",
        "properties": {
          ...
        },
        "required": []
      }
```

Related to #366

**Comment by unredundant** [comment]
Could you post the data class and an example response you are trying to create where you are seeing the double nullable? I believe the wrapping nullable should only be present if you have a response type of `responseType<MyEnum?>()`

**Comment by unredundant** [comment]
Ok i did some poking around w/ this.  The problem is that Kompendium takes whatever the first occurrance is type wise 

So if it sees `MyType?`, then all responses of this type will be marked as optional. Conversely, if it sees `MyType`, future `MyType?` responses will not show up as expected.  

So I think your PR is on the right track w/ this, we need to remove the optional wrapper from the schema handler, but we _do_ need it at some point in the pipeline, to be able to mark top level types as nullable.

**Comment by geirsagberg** [comment]
Gotcha. I updated the PR and added a new test, top level nullable types should be supported now :)

**Comment by unredundant** [comment]
closed by #368 

Thanks again!
